File: PC_App/make_header.js
const fs = require('fs');
const html = fs.readFileSync('../index.html', 'utf8');
const escaped = html.replace(/\\/g, '\\\\').replace(/"/g, '\\"').replace(/\r?\n/g, '\\n" \n"');
const content = `#pragma once\nconst char index_html[] PROGMEM = "${escaped}";\n`;
fs.writeFileSync('index_html.h', content);
console.log('Created index_html.h');

File: PC_App/make_header.py
import sys

with open('../index.html', 'r', encoding='utf-8') as f:
    html = f.read()

escaped = html.replace('\\', '\\\\').replace('"', '\\"').replace('\n', '\\n"\n"')

with open('index_html.h', 'w', encoding='utf-8') as f:
    f.write('#pragma once\nconst char index_html[] PROGMEM = "')
    f.write(escaped)
    f.write('";\n')

print("Created index_html.h")